### PR TITLE
Proposal: Samples: remove TEST_FAIL_ON_ERROR macro

### PR DESCRIPTION
**Pull request by antkmsft**
This is a continuation of an idea I had in comments for https://github.com/Azure/azure-sdk-for-c/pull/757.

### CODE REVIEW

**Comment by antkmsft** [comment]
@gilbertw   For some reason, I am getting all the github feedback notifications with delay today.
Ok, I will rename the macro in the original PR. What about returning non-0 code when built with az_nohttp? Leave the sample to exit with 0 as today, or switch to use RETURN_IF_FAILED?

File: sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c [hunk 1]
```diff
@@ -125,36 +125,44 @@ int main()
...
+  }
 
   printf("\n\n*********************************\nGet Key result: \n%s", response_buffer);
 
   /****************** 7) get key version from response ****/
   // version is still at http_response. Let's copy it to a new buffer
-  uint8_t copy_version_buf[40];
-  az_span copy_version_builder = AZ_SPAN_FROM_BUFFER(copy_version_buf);
+  uint8_t copy_version_buf[40] = { 0 };
+  az_span copy_version_buffer = AZ_SPAN_FROM_BUFFER(copy_version_buf);
```
Comment (ahsonkhan) [inline sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c:159]: nit:
```suggestion
  az_span copy_version_span = AZ_SPAN_FROM_BUFFER(copy_version_buf);
```

File: sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c [hunk 2]
```diff
@@ -125,36 +125,44 @@ int main()
...
+    return 6;
+  }
 
   printf("\n\n*********************************\nGet Key result: \n%s", response_buffer);
 
   /****************** 7) get key version from response ****/
   // version is still at http_response. Let's copy it to a new buffer
-  uint8_t copy_version_buf[40];
-  az_span copy_version_builder = AZ_SPAN_FROM_BUFFER(copy_version_buf);
+  uint8_t copy_version_buf[40] = { 0 };
```
Comment (ahsonkhan) [inline sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c:158]: nit: maybe naming the array and the span distinctly would be clearer.
```suggestion
  uint8_t copy_version_buffer[40] = { 0 };
```

**Comment by ahsonkhan** [comment]
> What about returning non-0 code when built with az_nohttp? Leave the sample to exit with 0 as today, or switch to use RETURN_IF_FAILED?

I think that is an error case, and it would be useful for the reader to know that something went wrong. So, returning a non-0 code makes sense to me.

File: sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c [hunk 3]
```diff
@@ -56,18 +56,6 @@
...
-  do \
-  { \
-    if (az_failed(result)) \
-    { \
-      printf("\n"); \
-      printf(message); \
-      printf("\n"); \
-      return 1; \
-    } \
-  } while (0)
```
Comment (vhvb1989) [inline sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c:69]: I see a lot of use of having this MACRO.
Formatting error message for all places, and also easy way to introduce a clean up routine. (will be needed for libcurl)

I think explicitly adding all the `if () else () ` statements makes the sample grow bigger and distract customer from the main idea of the sample  by looking all of that duplicated code

**Comment by gilbertw** [comment]
> @gilbertw For some reason, I am getting all the github feedback notifications with delay today.
> Ok, I will rename the macro in the original PR. What about returning non-0 code when built with az_nohttp? Leave the sample to exit with 0 as today, or switch to use RETURN_IF_FAILED?

It should return a non-zero on failure.  Doing that in the macro is reasonable, you could add a value, or you could simply return the az_result from the macro.

**Comment by ahsonkhan** [comment]
> or you could simply return the az_result from the macro.

I like that approach. There is little value coming up with arbitrary new exit codes on top of az_result.

**Comment by antkmsft** [comment]
Thank you, I will apply your feedback in new commits to #757.

**Pull request closed without merging**